### PR TITLE
Support new blinded block creation

### PR DESCRIPTION
**Pull request by rolfyone**
As part of this, the async handler had to be fixed to return a body, it was previously trying to serialize its own optional.

Reworked the old block creation so that we don't have 2 sets of code.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java [hunk 1]
```diff
@@ -125,9 +129,15 @@ public boolean isStoreAvailable() {
     if (currentSlot.isGreaterThan(slot)) {
       throw new IllegalArgumentException(CANNOT_PRODUCE_HISTORIC_BLOCK);
     }
+    return validatorApiChannel.createUnsignedBlock(slot, randao, graffiti, false);
```
Comment (tbenr) [inline data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java]: why don't you pass the `isBlinded` param?

**Review by tbenr — APPROVED** [review]
LGTM with a couple of question

File: data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java [hunk 2]
```diff
@@ -125,9 +129,15 @@ public boolean isStoreAvailable() {
...
+  }
 
-    return validatorApiChannel
-        .createUnsignedBlock(
+  public SafeFuture<Optional<BeaconBlock>> getUnsignedBeaconBlockAtSlot(
+      UInt64 slot, BLSSignature randao, Optional<Bytes32> graffiti) {
+    if (randao == null) {
+      throw new IllegalArgumentException(NO_RANDAO_PROVIDED);
```
Comment (tbenr) [inline data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java:138]: can we remove this check? it is done in the called method in any case, right?

**Review by tbenr — APPROVED** [review]

File: data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java (hunk 2 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java:138]: i started without it but then the conversion causes issues, so putting this back was the easy option...

File: data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java (hunk 1 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java]: ah i should really, thats where i started and just a plain old bug

File: data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java (hunk 2 above)
Comment (tbenr) [reply data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java:138]: yeah, true.

**Pull request merged**